user_service/main.go

```diff
@@ -0,0 +1,181 @@
+	}
+
+	if user.Password != req.Password {
+		log.Printf("%v != %v", user.Password, req.Password)
+		return nil, fmt.Errorf("invalid password")
+	}
+
+	// token, err := generateRandomToken()
+
+	if err != nil {
+		return nil, fmt.Errorf("can't generate token")
+	}
+
+	return &usermanagement.LoginUserResponse{
+		Token: "token",
+	}, nil
+}
+
+func (s *server) GetUser(ctx context.Context, in *usermanagement.GetUserRequest) (*usermanagement.GetUserResponse, error) {
+	user, err := s.usersRepo.FindByUsername(ctx, in.UserId)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("Received: %v", in.UserId)
+	return &usermanagement.GetUserResponse{
+		User: &usermanagement.User{
+			UserId: user.ID,
+			Name:   user.Name,
+		},
+	}, nil
+}
+
+func (s *server) StoreUser(ctx context.Context, req *usermanagement.StoreUserRequest) (*usermanagement.StoreUserResponse, error) {
+	log.Printf("Received store user request for user: %v", req.User.Name)
+
+	user := user.User{
+		ID:       req.User.UserId,
+		Name:     req.User.Name,
+		Password: req.User.Password,
+	}
+
+	err := s.usersRepo.Save(ctx, user)
+	if err != nil {
+		return &usermanagement.StoreUserResponse{Result: "Failed to store user"}, err
+	}
+	return &usermanagement.StoreUserResponse{Result: "User stored successfully!!!"}, nil
+}
+
+// Global variable to hold the image data.
+var imageData []byte
+
+func (s *server) UploadImage(stream usermanagement.UserService_UploadImageServer) error {
+	log.Printf("Received upload image request")
+	for {
+		// Receive a message from the stream.
+		req, err := stream.Recv()
+		if err == io.EOF {
+			// The client has finished sending data.
+			break
+		}
+		if err != nil {
+			return status.Errorf(codes.Unknown, "failed to receive data: %v", err)
+		}
+
+		// Append the received data to the global variable.
+		imageData = append(imageData, req.Data...)
+	}
+
+	// Send a response to the client.
+	res := &usermanagement.UploadImageResponse{
+		Result: "Image uploaded successfully",
+	}
+	if err := stream.SendAndClose(res); err != nil {
+		return status.Errorf(codes.Unknown, "failed to send response: %v", err)
+	}
+
+	return nil
+}
+
+func (s *server) FetchImage(req *usermanagement.FetchImageRequest, stream usermanagement.UserService_FetchImageServer) error {
+	log.Printf("Received fetch image request")
+
+	chunkSize := 1024 // Size of each chunk in bytes
+	for i := 0; i < len(imageData); i += chunkSize {
+		end := i + chunkSize
+		if end > len(imageData) {
+			end = len(imageData)
+		}
+
+		// Create ImageData from the chunk.
+		data := &usermanagement.ImageData{
+			Data: imageData[i:end],
+		}
+
+		// Send the chunk to the client.
+		if err := stream.Send(data); err != nil {
+			return status.Errorf(codes.Unknown, "failed to send data: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	portStr := os.Getenv("GRPC_PORT")
+	if portStr == "" {
+		portStr = "50051" // default port
+	}
+	port, err := strconv.Atoi(portStr)
```

Comment:
sorry, but why you convert a string to an integer only to later convert it back to a string on line 143?